src/main/scala/zio/prelude/AnnihilatingZero.scala

```diff
@@ -0,0 +1,81 @@
+package zio.prelude
+
+import zio.prelude.coherent.AnnihilatingZeroEqual
+import zio.prelude.newtypes.{ Prod, Sum }
+import zio.test.TestResult
+import zio.test.laws.{ Lawful, Laws }
+
+trait AnnihilatingZero[A, +Addition[x] <: Identity[x], +Multiplication[x] <: Associative[x]]
```

Comment:
In Prelude, `Identity` is used for the identity law, and `identity` for the identity element.

This suggests, perhaps, that `Annihilation` would be the name of this type class, and `annihilation` the name of the "anihilator" element.

Reply:
Good idea! :+1:

Reply:
Renamed :heavy_check_mark: